File: server/models/Contacts.js
const mongoose = require('mongoose');

const contactSchema = mongoose.Schema({
    userId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    contactUserId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    contactPhoneNumber: {
        type: String,
        required: true
    },
    contactName: {
        type: String,
        required: true
    },
    addedAt: {
        type: Date,
        default: Date.now
    },
    isBlocked: {
        type: Boolean,
        default: false
    }
}, {
    timestamps: true
});

// Ensure a user can't add the same contact twice
contactSchema.index({ userId: 1, contactUserId: 1 }, { unique: true });

const Contacts = mongoose.model('Contact', contactSchema);

module.exports = Contacts;

File: go_server/models/contacts.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Contact struct {
	ID                 primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	UserID             primitive.ObjectID `bson:"userId" json:"userId"`
	ContactUserID      primitive.ObjectID `bson:"contactUserId" json:"contactUserId"`
	ContactPhoneNumber string             `bson:"contactPhoneNumber" json:"contactPhoneNumber"`
	ContactName        string             `bson:"contactName" json:"contactName"`
	AddedAt            time.Time          `bson:"addedAt,omitempty" json:"addedAt"`
	IsBlocked          bool               `bson:"isBlocked,omitempty" json:"isBlocked"`
	CreatedAt          time.Time          `bson:"createdAt,omitempty" json:"createdAt"`
	UpdatedAt          time.Time          `bson:"updatedAt,omitempty" json:"updatedAt"`
}